common/src/main/java/com/wynntils/sockets/model/SocketModel.java

```diff
@@ -0,0 +1,202 @@
+import com.wynntils.sockets.SocketClientHandler;
+import com.wynntils.sockets.events.SocketEvent;
+import com.wynntils.sockets.objects.PlayerStatus;
+import com.wynntils.wynn.event.CharacterUpdateEvent;
+import com.wynntils.wynn.event.RelationsUpdateEvent;
+import com.wynntils.wynn.event.WorldStateEvent;
+import com.wynntils.wynn.model.ActionBarModel;
+import com.wynntils.wynn.model.CharacterManager;
+import com.wynntils.wynn.model.WorldStateManager;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import net.minecraft.client.player.LocalPlayer;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class SocketModel extends Model {
+    private static final int TICKS_PER_UPDATE = 5;
+    private static final int MS_PER_PING = 1000;
+
+    private static HadesConnection hadesConnection;
+    private static int tickCountUntilUpdate = 0;
+    private static PlayerStatus lastSentStatus;
+    private static ScheduledExecutorService pingScheduler;
+
+    public static void init() {
+        tryCreateConnection();
+    }
+
+    public static void disable() {
+        tryDisconnect();
+    }
+
+    private static void tryCreateConnection() {
+        if (WebManager.getAccount().isEmpty()) {
+            WynntilsMod.error("Cannot connect to HadesServer when WebManager does not have account.");
+            return;
+        }
+
+        try {
+            hadesConnection = new HadesNetworkBuilder()
+                    .setAddress(InetAddress.getByName("io.wynntils.com"), 9000)
+                    .setDirection(PacketDirection.SERVER)
+                    .setCompressionThreshold(256)
+                    .setHandlerFactory(SocketClientHandler::new)
+                    .buildClient();
+
+            tickCountUntilUpdate = 0;
+            lastSentStatus = null;
+        } catch (UnknownHostException e) {
+            WynntilsMod.error("Could not resolve Hades host address.", e);
+        }
+    }
+
+    private static void tryDisconnect() {
+        if (hadesConnection != null && hadesConnection.isOpen()) {
+            hadesConnection.disconnect();
+        }
+    }
+
+    @SubscribeEvent
+    public static void onAuth(SocketEvent.Authenticated event) {
+        pingScheduler = Executors.newSingleThreadScheduledExecutor();
+        pingScheduler.scheduleAtFixedRate(SocketModel::sendPing, 0, MS_PER_PING, TimeUnit.MILLISECONDS);
+    }
+
+    @SubscribeEvent
+    public static void onDisconnect(SocketEvent.Disconnected event) {
+        pingScheduler.shutdown();
+    }
+
+    private static void sendPing() {
+        if (!isSocketOpen()) return;
+
+        hadesConnection.sendPacketAndFlush(new HCPacketPing(System.currentTimeMillis()));
+    }
+
+    @SubscribeEvent
+    public static void onFriendListUpdate(RelationsUpdateEvent.FriendList event) {
```

Comment:
SocketModel implicitly depends on PlayerRelationsModel here, don't see a resolution besides moving code to SocketFeature

Reply:
Or you just make sure a feature that depends on socket model also depends on relations model

Reply:
We could have models depend on other models :)

Reply:
Ew. I think we should do what magicus wanted and merge all models into one and make them non-disablable.

Reply:
I'm sorry I won't be able to work on that (at least for a very long time). But I whole-heartedly will support if you do it! :)